pkg/cover/html.go

```diff
@@ -24,40 +24,46 @@ import (
 	"github.com/google/syzkaller/sys/targets"
 )
 
-func fixUpPCs(target string, progs []Prog, coverFilter map[uint32]uint32) []Prog {
-	if coverFilter != nil {
-		for _, prog := range progs {
+func (rg *ReportGenerator) FixupPCs(target *targets.Target, progs []Prog, coverFilter map[uint32]uint32) []Prog {
+	nProgs := make([]Prog, len(progs))
+	copy(nProgs, progs)
+
+	// On arm64 as PLT is enabled by default, .text section is loaded after .plt section,
+	// so there is X bytes offset from module load address for .text section
+	// we need to remove the X bytes offset in order to correct module symbol address.
+	// On some chip, the offset is 0x18 while some newer chip is 0x24,
+	// so for linux, auto detects the offset by subtract module load address from first loaded symbol in module.
+	if target.Arch == targets.ARM64 && target.ModuleLoadOffset != 0 {
```

Comment:
The check for ARM64 looks excessive. What does ModuleLoadOffset contain for other arches? Where is it used for other arches?
If it's the only place where it's used, then it's better to store 0 in ModuleLoadOffset when we don't need to do any offsetting.

OS/arch-independent code is always better, simpler to review and reason about.